### PR TITLE
Add README and CONTRIBUTING files

### PR DESCRIPTION
**Pull request by KotlinIsland**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,40 +18,23 @@ body) of requests. It also does not aim to be the fastest, or most secure.
...
+ProxyAuth does not have a GUI, and should be run from a console / shell such as `bash` / `cmd.exe`.
 
-`C:\example\ProxyAuth\src\main\java>javac -d ..\bin proxyauth\*.java`
+### Windows
 
-## Installation
+`java -jar ProxyAuth-0.1.0.jar`
 
-TODO - depends on https://github.com/Zeckie/ProxyAuth/issues/6 (configuration)
+### Linux/MaxOS
```
Comment (Zeckie) [inline README.md:31]: Is MaxOS a type of Unix, like [macOS](https://en.wikipedia.org/wiki/MacOS)?

File: README.md [hunk 2]
```diff
@@ -18,40 +18,23 @@ body) of requests. It also does not aim to be the fastest, or most secure.
...
 
-ProxyAuth does not have a GUI, and should be run from a console / shell such as `cmd.exe`.
-`java -jar ProxyAuth-0.1.0.jar`
+`ProxyAuth-0.1.0.jar`
```
Comment (Zeckie) [inline README.md:33]: Do most Linux and MaxOS installations support this out of the box (without having to install extra tools such as `binfmt-support`)?

File: README.md [hunk 3]
```diff
@@ -18,40 +18,23 @@ body) of requests. It also does not aim to be the fastest, or most secure.
...
 
 ## Licence
 
-[GNU GPLv3](LICENCE) - https://www.gnu.org/licenses/gpl-3.0.html
+[GNU GPLv3](LICENSE) - https://www.gnu.org/licenses/gpl-3.0.html
```
Comment (Zeckie) [inline README.md:37]: I was going to fix the spelling of the filename, but I guess GitHub probably only works with the USA spelling

File: README.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ The typical use case is inside an organization that requires users to authentica
 
 Some applications need proxy settings (including password) to be set in
 [non-standardized](https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/)  environment variables such
-as `http_proxy`,
+as `HTTP_PROXY`,
```
Comment (Zeckie) [inline README.md:10]: @KotlinIsland KotlinIsland [13 days ago](https://github.com/Zeckie/ProxyAuth/pull/13#pullrequestreview-787650038)
> Unofficial standard naming of these variables is lowercase.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,8 +1,38 @@
...
+
+### Without Gradle
+
+It does not rely on any third-party libraries, so can be built using any Java IDE, or even javac.
+
+`C:\example\ProxyAuth\src\main\java>javac -d ../bin proxyauth/*.java`
+
+## Design
+
+The main language of the project is Java (NOT Kotlin!).
```
Comment (Zeckie) [inline CONTRIBUTING.md:33]: I agree. We probably should go further and mandate Java for all code that can be written in Java.

File: README.md (hunk 2 above)
Comment (Zeckie) [reply README.md:33]: Which OSs / distros have you tested this on?

File: README.md [hunk 6]
```diff
@@ -18,40 +18,23 @@ body) of requests. It also does not aim to be the fastest, or most secure.
 
 Java (version 16 or later). No third-party libraries required for execution.
 
-Dependencies for build / test (such as JUnit) are managed using Gradle. All dependencies should be current, stable
-versions.
+## Running
 
-## Building
```
Comment (Zeckie) [inline README.md:24]: So only contributors will want to build it?

File: README.md (hunk 3 above)
Comment (KotlinIsland) [reply README.md:31]: I would think that the us spelling would be better here as things looking for a license file would assume us spelling.

File: README.md (hunk 6 above)
Comment (KotlinIsland) [reply README.md:24]: I don't see why an end user would want to build it. Unless they wanted to customize the behavior.

File: CONTRIBUTING.md (hunk 5 above)
Comment (KotlinIsland) [reply CONTRIBUTING.md:41]: We could ditch Gradle and include a jar that builds the project instead.

File: README.md (hunk 2 above)
Comment (KotlinIsland) [reply README.md]: hmm, I don't know where I saw this happening. yeah, this isn't a good one.

File: README.md [hunk 7]
```diff
@@ -18,41 +18,20 @@ body) of requests. It also does not aim to be the fastest, or most secure.
...
-`C:\example\ProxyAuth\src\main\java>javac -d ..\bin proxyauth\*.java`
-
-## Installation
+## Running
 
 TODO - depends on https://github.com/Zeckie/ProxyAuth/issues/6 (configuration)
 
-ProxyAuth does not have a GUI, and should be run from a console / shell such as `cmd.exe`.
-`java -jar ProxyAuth-0.1.0.jar`
+ProxyAuth does not have a GUI, and should be run from a console / shell such as `bash` / `powershell`.
```
Comment (Zeckie) [inline README.md:25]: powershell - is that a joke?

File: CONTRIBUTING.md (hunk 5 above)
Comment (Zeckie) [reply CONTRIBUTING.md:41]: https://stackoverflow.com/questions/12766808/why-is-javac-source-code-is-written-in-java

File: README.md (hunk 6 above)
Comment (Zeckie) [reply README.md:24]: Apparently, there are some people who like running applications before they are released.

Comment (KotlinIsland) [reply README.md:24]: Host pre-releases then. If I wanted to know how to build/use from a main branch I would look in contributing.

But w/e it's not really a big deal.

File: README.md (hunk 7 above)
Comment (KotlinIsland) [reply README.md:25]: I think cmd isn't being developed anymore, (https://www.theregister.com/2020/05/27/windows_cmd_maintenance_mode/), it's more like a legacy tool.

Comment (Zeckie) [reply README.md:25]: > Cmd is one of the most frequently run executables on Windows ...

> The Windows Cmd / Command-Line shell is NOT being removed from Windows in the near or distant future!

https://devblogs.microsoft.com/commandline/rumors-of-cmds-death-have-been-greatly-exaggerated/

Also, looks like Gradle Wrapper still uses cmd.exe (`gradlew.bat` not `gradlew.ps1`)

Comment (Zeckie) [reply README.md:25]: I've updated readme to mention powershell

**Comment by Zeckie** [comment]
Most of the suggestions from this PR have been integrated into #13

**Pull request closed without merging**